### PR TITLE
[T4A4][F10-B2] Shi Xiyue 

### PR DESCRIPTION
**Pull request by shixiyue**
Ready for review

### CODE REVIEW

**Comment by shixiyue** [comment]
Review in tutorial

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -6,11 +6,12 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address implements Printable{
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    private static final String TYPE = " Address: ";
```
Comment (KulaniT) [inline src/seedu/addressbook/data/person/Address.java:14]: why public?

File: src/seedu/addressbook/data/person/ReadOnlyPerson.java [hunk 2]
```diff
@@ -82,4 +67,16 @@ default String getAsTextHidePrivate() {
...
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    default String getPrintableString(Printable... printables){
+    	final StringBuilder builder = new StringBuilder();
+    	for (Printable printable: printables){
+    		builder.append(printable.getPrintableString());
+    	}
+    	return builder.toString();
+    }
+
```
Comment (KulaniT) [inline src/seedu/addressbook/data/person/ReadOnlyPerson.java:81]: Good work

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -55,4 +56,16 @@ public int hashCode() {
     public boolean isPrivate() {
         return isPrivate;
     }
+
+	@Override
+	public String getPrintableString() {
+		final StringBuilder builder = new StringBuilder();
```
Comment (KulaniT) [inline src/seedu/addressbook/data/person/Address.java:62]: Indentation

**Comment by KulaniT** [comment]
@shixiyue  : Good try implementing ISP. Please close the PR after reading comments

Practice to give the lowest possible access modifier and to add comments

File: src/seedu/addressbook/data/person/Address.java (hunk 1 above)
Comment (shixiyue) [reply src/seedu/addressbook/data/person/Address.java:14]: Ok I see. Thanks! I will practice to give the lowest possible access modifier.

**Pull request closed without merging**